azure-log-analytics/operations.py

```diff
@@ -6,56 +6,82 @@
 
 from requests import request, exceptions as req_exceptions
 from .microsoft_api_auth import *
+from connectors.core.connector import get_logger, ConnectorError
+from connectors.core.utils import update_connnector_config
 
 logger = get_logger('azure-log-analytics')
 
-loganalysis_api_endpoint = 'https://api.loganalytics.io/v1'
-analytics_api_endpoint = 'https://management.azure.com'
+MANAGE_SERVER_URL = 'https://management.azure.com'
+MANAGE_API_VERSION = '2020-08-01'
+LOG_API_VERSION = '2022-10-27_Preview'
 
 
-def api_request(method, endpoint, connector_info, config, params=None, data=None, headers={}):
-    try:
-        ms = MicrosoftAuth(config)
-        token = ms.validate_token(config, connector_info)
-        headers['Authorization'] = token
-        headers['Content-Type'] = 'application/json'
-        logger.debug("Endpoint: {0}".format(endpoint))
+class AzureLogAnalytics(object):
+    def __init__(self, config):
+        self.server_url = 'https://api.loganalytics.io'
```

Comment:
Do we really need to hardcode this URL here or we can take this from config? We can set this URL as a default.

Reply:
We are using two base url to execute this connector actions.

https://management.azure.com -> Saved Search actions
https://api.loganalytics.io -> Execute Query action

We don't have any default url for this connector.